### PR TITLE
Feat: add KUBSAN

### PR DESCRIPTION
**Pull request by Abb1x**
I ported tinyubsan to brutal :)

### CODE REVIEW

File: sources/kernel/tinyubsan.c [hunk 1]
```diff
@@ -0,0 +1,173 @@
+#include <brutal/log.h>
```
Comment (sleepy-monax) [inline sources/kernel/tinyubsan.c:1]: This file should live in libc/ubsan

File: sources/kernel/tinyubsan.c [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+#include <brutal/text.h>
+#include <stdint.h>
```
Comment (sleepy-monax) [inline sources/kernel/tinyubsan.c:3]: You should include <brutal/base/std.h>

File: sources/kernel/tinyubsan.c [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+#define tu_print(...) panic(__VA_ARGS__)
```
Comment (sleepy-monax) [inline sources/kernel/tinyubsan.c:4]: We can just CTRL-F CTRL-R tu_print by panic

File: sources/kernel/tinyubsan.c [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+struct tu_source_location
```
Comment (sleepy-monax) [inline sources/kernel/tinyubsan.c:6]: The prefix should be ubsan_

**Review by sleepy-monax — CHANGES_REQUESTED** [review]
Nice PR we just need to fix these 4 comments

File: sources/kernel/tinyubsan.c [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+struct tu_invalid_builtin_data
+{
+    struct tu_source_location location;
+    unsigned char kind;
+};
+
+static void tu_print_location(char *message, struct tu_source_location loc)
+{
+  tu_print("tinyubsan: {} at file {}, line {}, column {}", message, str_cast(loc.file), loc.line, loc.column);
```
Comment (Supercip971) [inline sources/kernel/tinyubsan.c:83]: rip l'alignement

**Review by Supercip971 — CHANGES_REQUESTED** [review]
code format fix

+ thank you <3

**Comment by sleepy-monax** [comment]
LGTM 👍

**Pull request merged**